Source/WebCore/css/CSSFontSelector.cpp

```diff
@@ -445,4 +446,19 @@ RefPtr<Font> CSSFontSelector::fallbackFontAt(const FontDescription& fontDescript
     return font;
 }
 
+bool CSSFontSelector::isSimpleFontSelectorForDescription() const
+{
+    // font face rules still pending
+    if (m_stagingArea.size())
+        return false;
+
+    // FIXME: remove this when we fix counter style rules mutation.
+    if (auto* document = dynamicDowncast<Document>(m_context.get())) {
+        if (document->counterStyleRegistry().hasAuthorCounterStyles())
+            return false;
+    }
+
+    return !m_cssFontFaceSet->faceCount() && m_featureValues.isEmpty() && m_paletteMap.isEmpty();
```

Comment:
I wonder if putting the `faceCount()` call that involves a pointer dereference to the end of the conditionals would squeeze any more perf out since the local container empty checks would fail the && operation immediately

Reply:
This is a good point. My reasoning here (based only on intuition and no data) was that @font-face is more common than `m_featureValues` and `m_paletteMap`, which are usually empty. So if we see an @font-face rule we would already return false.

I'd imagine for SP3 all of these are usually empty but that's something we can easily measure on an A/B.

Reply:
We could early return separately for fontFace, featureValues and paletteMap to make it slightly more readable and certainly slightly faster.

The counter style fix should be the last one also (certainly the most costly).